src/util-logopenfile.c

```diff
@@ -592,26 +863,35 @@ static int  LogFileWriteRedis(LogFileCtx *file_ctx, const char *string, size_t s
             file_ctx->redis_setup.batch_count++;
         }
     } else {
-        redisReply *reply = redisCommand(file_ctx->redis, "%s %s %s",
+        redisReply *reply = redisCommand(redis, "%s %s %s",
                 file_ctx->redis_setup.command,
                 file_ctx->redis_setup.key,
                 string);
 
-        switch (reply->type) {
-            case REDIS_REPLY_ERROR:
-                SCLogWarning(SC_ERR_SOCKET, "Redis error: %s", reply->str);
-                SCConfLogReopenRedis(file_ctx);
-                break;
-            case REDIS_REPLY_INTEGER:
-                SCLogDebug("Redis integer %lld", reply->integer);
-                break;
-            default:
-                SCLogError(SC_ERR_INVALID_VALUE,
-                        "Redis default triggered with %d", reply->type);
+        /* We may lose the reply if disconnection happens*/
+        if (reply)  {
```

Comment:
Could you please put this part in a separate commit as this is a fix we need to backport to stable.